### DISCUSSION TITLE
DateTime直接入库天数会加一天？

### DISCUSSION THREAD

**Discussion by HKTKJung** [looly/hutool#2249] [Q&A]
最近遇到DateTime直接入库天数会加一天
DateTime不能直接入库吗？一定要toJdkDate入库才没问题？
mysql 5.7.24

**Comment by looly** [comment]
我猜测你获取的日期中毫秒是999，导致存入问题。

见：https://github.com/dromara/hutool/issues/2025

**Reply by HKTKJung** [reply]
endOfDay毫秒是999这个确实也遇到了，但是这个影响不大，我遇到的是
入库前：2022-04-11 14:36:56
入库后：2022-04-12 14:36:56
毫秒debug也不是999

**Reply by looly** [reply]
奇怪了啊。

你看下数据库中的日期是不是只有Date？然后根据时间做了四舍五入？

尝试使用DateUtil.beginOfDay获取值传入试下。

**Reply by HKTKJung** [reply]
DateUtil.date获取时间，然后存入数据库的，没做什么处理，数据库也是datetime，debug前还是正确的，已入库天数就加1了，时间部分没问题
还有，我今天试了下正常了，复现不出来。。。

**Reply by looly** [reply]
那就需要检查你的代码了。

如果复现问题，麻烦提个issue。
